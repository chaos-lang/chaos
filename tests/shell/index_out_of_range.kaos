list a = [1,2,3]
a[0]
a[1]
a[2]
a[3]
a[123400]
a[-1]
a[-2]
a[-3]
a[-4]
a[-123400]

str b = "mike"
b[0]
b[1]
b[2]
b[3]
b[4]
b[123400]
b[-1]
b[-2]
b[-3]
b[-4]
b[-5]
b[-123400]

// Tests related to deleted index
bool list f = [false]
del f[0]
f
num f = 5
f[0] = true
f

// Complex element to complex element assignment on itself
del a
list a = [1, 2]
a
a[1] = a[2]
a
