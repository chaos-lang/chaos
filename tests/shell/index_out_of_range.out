1
2
3
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 5
Index out of range: 3 for list: a
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
Index out of range: 123400 for list: a
Absorbed by Interactive Shell
3
2
1
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 4
Index out of range: -4 for list: a
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
Index out of range: -123400 for list: a
Absorbed by Interactive Shell
m
i
k
e
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 7
Index out of range: 4 for string: b
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
Index out of range: 123400 for string: b
Absorbed by Interactive Shell
e
k
i
m
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 5
Index out of range: -5 for string: b
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
Index out of range: -123400 for string: b
Absorbed by Interactive Shell
[]
[start] =>
	{id: 0, name: a, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 4, 2nd_type: 3, value_type: 4, role: 0, param_of: } =>
	{id: 1, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 2, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 3, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 14, name: b, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 16, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 18, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 20, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 22, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 26, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 28, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 30, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 32, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 35, name: f, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 4, 2nd_type: 0, value_type: 4, role: 0, param_of: } =>
[end]
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 7
The variable name 'f' is already defined!
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
Index out of range: 0 for list: f
Absorbed by Interactive Shell
[]
[start] =>
	{id: 0, name: a, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 4, 2nd_type: 3, value_type: 4, role: 0, param_of: } =>
	{id: 1, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 2, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 3, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 14, name: b, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 16, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 18, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 20, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 22, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 26, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 28, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 30, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 32, name: (null), 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 35, name: f, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 4, 2nd_type: 0, value_type: 4, role: 0, param_of: } =>
[end]
[1, 2]
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 8
Index out of range: 2 for list: a
Absorbed by Interactive Shell
